### PR TITLE
Use Rails 7

### PR DESCRIPTION
**Pull request by dmarcoux**
That's it, we're there now. Let's do it! :slightly_smiling_face:

### CODE REVIEW

**Comment by dmarcoux** [comment]
[Update Rails to version 7](https://trello.com/c/buZsU7Yh/2017-update-rails-to-version-7)

**Comment by hennevogel** [comment]
showtime

File: src/api/db/schema.rb [hunk 1]
```diff
@@ -100,41 +99,41 @@
   create_table "backend_infos", id: :integer, charset: "utf8mb4", collation: "utf8mb4_unicode_ci", options: "ENGINE=InnoDB ROW_FORMAT=DYNAMIC", force: :cascade do |t|
     t.string "key", null: false, collation: "utf8mb3_unicode_ci"
     t.string "value", null: false, collation: "utf8mb3_unicode_ci"
-    t.datetime "created_at"
-    t.datetime "updated_at", precision: 6
+    t.datetime "created_at", precision: nil
+    t.datetime "updated_at"
```
Comment (hennevogel) [inline src/api/db/schema.rb:103]: The spec is failing with this or do you just do those out of precaution?

Comment (dmarcoux) [reply src/api/db/schema.rb:103]: I have done this to follow the [Upgrading Ruby on Rails guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#rails-version-is-now-included-in-the-active-record-schema-dump).

**Comment by krauselukas** [comment]
DO NOT MERGE: We first want to test in staging

**Review by eduardoj — APPROVED** [review]

**Comment by dmarcoux** [comment]
We plan to deploy this during the maintenance window tomorrow on 03.11.2022.

**Pull request merged**